le_dash/banner.py

```
@@ -1,18 +1,37 @@
 import six
+import hashlib
 import requests
 from django.conf import settings
+from django.core.cache import cache
+import logging
 
 if six.PY2:
     from urlparse import urljoin
 else:
     from urllib.parse import urljoin
 
 
+logger = logging.getLogger(__name__)
+
+
 def banner_req(url, params):
+    #  md5 is cheaper than urlunparse/urlencode params
+    get_url = hashlib.md5("{0}".format(params))
     params['fmt'] = 'json'
+    try:
+        result = cache.get(get_url)
+        if result:
+            return result
+    except:
+        logger.error("Caching is broken")
     resp = requests.get(url, params)
     resp.raise_for_status()
-    return resp.json()
+    result = resp.json()
+    try:
+        cache.set(get_url, result)
+    except:
```

What could go wrong here? Is it really OK for it to silently fail?